src/components/Onboarding/Screens/HoneyswapLiquidity.js

```diff
@@ -1,19 +1,342 @@
-import React from 'react'
-import { useOnboardingState } from '@providers/Onboarding'
+import React, { useCallback, useState } from 'react'
+import {
+  Box,
+  DropDown,
+  GU,
+  LoadingRing,
+  TextInput,
+  textStyle,
+  useTheme,
+} from '@1hive/1hive-ui'
+import Header from '../kit/Header'
 import Navigation from '../Navigation'
+import useHNYPriceOracle from '@hooks/useHNYPriceOracle'
+import { useOnboardingState } from '@providers/Onboarding'
+
+import { fromDecimals, toDecimals } from '@utils/math-utils'
+import { BYOT_TYPE } from '../constants'
+import { getLocalTokenIconBySymbol } from '@utils/token-utils'
+import BigNumber from 'bignumber.js'
+
+const MIN_HNY_USD = 100
+const HNY_DENOMINATION = 0
+
+function HoneyswapLiquidity() {
+  const theme = useTheme()
+  const {
+    config,
+    onBack,
+    onConfigChange,
+    onNext,
+    step,
+    steps,
+  } = useOnboardingState()
+  const { honeyTokenLiquidity, tokenLiquidity } = config.liquidity
+  const [hnyPrice, hnyPriceLoading] = useHNYPriceOracle(toDecimals('1', 18))
+
+  const [denomination, setDenomination] = useState(HNY_DENOMINATION) // 0 HNY, 1 USD
+  const [denominatedAmount, setDenominatedAmount] = useState(
+    honeyTokenLiquidity ? fromDecimals(honeyTokenLiquidity, 18) : ''
+  )
+  const [tokenAmount, setTokenAmount] = useState(
+    tokenLiquidity ? fromDecimals(tokenLiquidity, config.tokens.decimals) : ''
+  )
+
+  const handleDenominatedAmountChange = useCallback(event => {
+    const newAmount = event.target.value
+    if (isNaN(newAmount)) {
```

Comment:
I think we should check for negative numbers as well

```suggestion
    if (isNaN(newAmount) || newAmount < 0) {
```